.github/workflows/continuous-documentation.yml

```diff
@@ -0,0 +1,99 @@
+# This workflow deploys documentation to Vercel for previewing PRs
+
+name: Continuous Documentation
+
+on:
+  pull_request:
+    paths:
+    - 'doc/**'
+    - 'examples/**'
+    - 'pygmt/**'
+    - 'README.rst'
+    - '.github/workflows/continuous-documentation.yml'
+  pull_request_target:
+    paths:
+    - 'doc/**'
+    - 'examples/**'
+    - 'pygmt/**'
+    - 'README.rst'
+    - '.github/workflows/continuous-documentation.yml'
+
+jobs:
+  vercel:
+    name: Deploy to Vercel
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+        with:
+          # fecth all history so that setuptools-scm works
+          fetch-depth: 0
+
+      # Setup Miniconda
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2.0.1
+        with:
+          activate-environment: pygmt
+          python-version: 3.9
+          channels: conda-forge
+          miniconda-version: "latest"
+
+      - name: Install dependencies
+        run: |
+          conda install gmt=6.1.1 numpy pandas xarray netcdf4 packaging \
+                        ipython myst-parser sphinx sphinx-copybutton \
+                        sphinx-gallery sphinx_rtd_theme=0.4.3
```

Comment:
Here I only install the packages required for documentation, to save the building time.